tt_metal/Glossary.md

```diff
@@ -0,0 +1,145 @@
+  - **Low-Level Kernels -** Kernels at the bottom of the software stack, these are for the system’s basic functions.
+- **Mantissa -** Floating point number that represents significant digits of the number.
+- **Memory Types -** Memory can be Interleaved or Sharded and can be further categorized into height, width, and block sharding.
+  - **Interleaved Memory -** L1 interleaved distributes across all L1 memory banks sequentially.
+  - **Sharded Memory -** L1 sharded lays out tensors based user provided shard specifications.
+      - **Height Sharding -** L1 Memory split across dimension 0.
+      - **Width Sharding -** L1 Memory split across dimension 1.
+      - **Block Sharding -** L1 Memory split across dimension 0 and dimension 1.
+- **Memory Barrier -** Parallelization technique where operations are required to complete before moving on to other operations.
+- **Mesh -** 2D chip architecture.
+- **Mixed Precision -** Optimization technique using floating-point types in a model to enhance performance and reduce memory usage. The following are different precision types:
+  - **Mixed Precision -** Combined use of numerical precisions in a computational method.
+  - **Half Precision -** FP16 data transfers.
+  - **Single Precision -** FP32 data transfers.
+  - **Double Precision -** FP64 data transfers.
+- **Multicast -** Allows multiple processors to function simultaneously and asynchronously.
+- **PCIe -** Link between the chip and the host machine.
+- **Printf -** Library function that formats text.
+- **Profiler -** A tool used to assess the performance of software or an operation. Provides data to optimize system performance.
+- **PyBuda -** A Python interface for TT-Buda.
+- **PyTorch -** Optimized tensor library.
+- **RISC-V -** Processors (Baby RISCVs) that run C/C++ kernels and dispatch instructions to compute and data movement engines.
+- **Semaphore -** A variable or abstract data type used for access control. All semaphores use atomic operations.
+- **Singlecast -** Allows processors to execute the same instruction on multiple data points simultaneously.
+- **Tensix Core -** Processor cores that make up Tensix processors. Tensix cores contain the following components:
+  - **Matrix Engine -** Performs matrix multiplication, elementwise, and dot product operations on small matrices (tiles) of shape 32x32 or similar.
+  - **Tile Math Engine -** Handles standard algebraic operations.
+  - **Vector Math Engine -** Handles vectorized kernels such as Top-k, Sort, and special functions like GELU, Exp, and Sqrt.
+- **Tensor -** Data structure that can hold data in multiple dimensions. Tensors can be organized in a Row-Major or Tiled Layout:
+  - **Row-Major Tensor Layout -** Each row of the tensor corresponds to a page in the buffer.
+  - **Tiled Tensor Layout -** Pages represented as 2D tiles rather than being confined in a row.
```

Comment:
Row major or tiled refers to how the data is stored. RM is contiguous, tiled is organized into tile-sized faces. I wouldn't bring the concept of pages into the description of tensors. It's a bit lower-levelled and described inaccurately here. Instead, maybe also mention the other things tensor have, such as memory config (interleaved vs. sharded, L1 vs. DRAM), dtype, shape, etc...